nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/TailFile.java

```diff
@@ -1184,7 +1191,15 @@ private boolean recoverRolledFiles(final ProcessContext context, final ProcessSe
                             // This is the same file that we were reading when we shutdown. Start reading from this point on.
                             rolledOffFiles.remove(0);
                             FlowFile flowFile = session.create();
-                            flowFile = session.importFrom(in, flowFile);
+                            try {
+                                ByteArrayOutputStream out = new ByteArrayOutputStream();
```

Comment:
This approach is going to buffer everything that gets read into the ByteArrayOutputStream. This could potentially be a huge amount of data. We want to be sure that we don't buffer that up. Instead, I would recommend an approach like:
```
try (final OutputStream out = session.write(flowFile)) {
  readLines(fis.getChannel(), ByteBuffer.allocate(65536), out, new CRC32(), reReadOnNul, true);
}
```
This allows us to write directly to the content repository instead of buffering the data in a ByteArrayOutputStream.

Reply:
oh, wow, I wasn't aware of this version of `session.write`. I was also a bit concerned about the buffering, thanks, will update the PR

Reply:
Pushed a new commit: I got rid of the BAOS, but used `session.write(FlowFile, OutputStreamCallback)` because we need the updated FF later.